doc/source/notebooks/advanced/varying_noise.pct.py

```diff
@@ -114,10 +115,16 @@ def variational_expectations(self, Fmu, Fvar, Y):
 def objective_closure():
     return - model.log_marginal_likelihood()
 
-opt = gpflow.optimizers.Scipy()
-opt.minimize(objective_closure,
-             model.trainable_variables,
-             options=dict(maxiter=ci_niter(1000)))
+natgrad = NaturalGradient(gamma=1.0)
+adam = tf.optimizers.Adam()
+
+from gpflow.utilities import set_trainable
+set_trainable(model.q_mu, False)
+set_trainable(model.q_sqrt, False)
+
+for _ in range(ci_niter(1000)):
+    adam.minimize(objective_closure, model.trainable_variables)
+    natgrad.minimize(objective_closure, [(model.q_mu, model.q_sqrt)])
```

Comment:
shouldn't natgrad come first?

Reply:
i have absolutely no idea. should it?

Reply:
also curious - if it should, why?

Reply:
I think NatGrad should come first, for the following motivation: in the conjugate case (Gaussian likelihood), a single natgrad step with gamma=1.0 brings the variational distribution q(u) to its optimum. The ELBO is only the right objective for variational parameters, whereas there might be a bias in the hyperparameters. If we bring the q(u) closer to its optimum, there will be better gradients for the adam optimization of the hyperparameters. In practice this would only make a difference for the first (& last) step but conceptually I think it's useful to revert the order here and make natgrad.minimize() come first.